nixos/modules/services/desktops/gnome3/evolution-data-server.nix

```diff
@@ -14,32 +14,54 @@ with lib;
 
   options = {
 
-    services.gnome3.evolution-data-server = {
-
-      enable = mkOption {
+    services.gnome3.evolution = {
+      enableDataServer = mkOption {
         type = types.bool;
         default = false;
         description = ''
           Whether to enable Evolution Data Server, a collection of services for
           storing addressbooks and calendars.
         '';
       };
+      enableGui = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to install Evolution, a Personal information management application that provides integrated mail, calendaring and address book functionality.
+        '';
+      };
+      plugins = mkOption {
+        type = types.listOf types.package;
+        default = [ ];
+        example = literalExample "[ pkgs.evolution-ews ]";
+        description = "Plugins from Evolution and its Data Server.";
+      };
 
     };
-
   };
 
+  imports = [
+    (mkRenamedOptionModule [ "services" "gnome3" "evolution-data-server" "enable" ] [ "services" "gnome3" "evolution" "enableDataServer" ])
+  ];
 
-  ###### implementation
-
-  config = mkIf config.services.gnome3.evolution-data-server.enable {
-
-    environment.systemPackages = [ pkgs.gnome3.evolution-data-server ];
-
-    services.dbus.packages = [ pkgs.gnome3.evolution-data-server ];
 
-    systemd.packages = [ pkgs.gnome3.evolution-data-server ];
-
-  };
+  ###### implementation
 
+  config =
+    let
+      plugins = config.services.gnome3.evolution.plugins ++ (lib.optional config.services.gnome3.evolution.enableGui pkgs.evolution);
+      bundle = pkgs.evolutionWithPlugins.override { inherit plugins; };
+    in
+    mkMerge [
+      (mkIf config.services.gnome3.evolution.enableDataServer {
+        environment.systemPackages = [ bundle ];
+
+        services.dbus.packages = [ bundle ];
+
+        systemd.packages = [ bundle ];
+      })
+      (mkIf config.services.gnome3.evolution.enableGui {
+        services.gnome3.evolution.enableDataServer = true;
+      })
```

Comment:
I'm not sure I really like these option names from a user pov, I feel like this would make more sense
```
programs.evolution.enable
services.gnome3.evolution-data-server.enable
```

Reply:
Done